pkgs/applications/emulators/uxn/default.nix

```diff
@@ -21,14 +21,14 @@ stdenv.mkDerivation rec {
 
   dontConfigure = true;
 
-  # It is easier to emulate build.sh script
+  patchPhase = ''
```

Comment:
```suggestion
  postPatch = ''
```

Reply:
why?

Reply:
**Tradition**. Just follow it, _no questions_.

-------------------------------

Okay, I know this is not a satisfactory answer. Here is the best technical answer I can provide:

The most correct way to use custom `phase`s is

 ```nix
buildPhase = ''
  runHook preBuild

   # insert here the custom code you want

  runHook postBuild
'';
 ``` 

This`runHook` decorating the custom code is because Nixpkgs can be monkey-patched by the end user, via overlays, flakes etc., and these `pre/post` hooks help such _composability_. 

-----------------------

The `patchPhase` has another peculiarity: 

Usually we (and many other distros, for that matter) provide custom unified diff `.patch` files that modify the source code. These patch files are listed on the `patch = [ XXX.patch YYY.patch ... ];` attribute, and the default  `patchPhase` basically runs `patch -p1 < XXX.patch` on them.

When you rewrite the whole `patchPhase` like this

 ```nix
patchPhase = ''
  runHook prePatch

   # custom patch code, usually some crazy sed/awk one-liner

  runHook postPatch
'';
 ```

you leave no option to apply such `.patch`es, forcing the next developer to write the `patch -p1 < ...` code, effectively duplicating the effort of the default `patchPhase`.

That being said, the best when you want to use some sed-awk dark magic to modify the upstream code is to write it as a `postPatch`.

-------------------------

Now do it.

Reply:
Thanks for the explanations.